rama-net/src/mode.rs

```diff
@@ -0,0 +1,43 @@
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+/// Enum representing the IP modes that can be used by the DNS resolver.
+pub enum IPModes {
```

Comment:
There are 2 enums required:

```
enum DnsResolveIpMode
    dual (default)
    singleIpV4
    singleIpV6
    dualPreferIpv4

enum ConnectIpMode
    dual (default)
    ipv4
    ipv6
```

The struct `DnsResolveIpMode` can be removed as there should be no need to wrap the resolver, this would be bad for ergonomics and doesn't really solve anything.

The difference between the 2 enums is that:

- ConnectIpMode is used only for establishing a connection
- DnsResolveIpMode is used only for resolving a domain
- these two are separate and do not influence each other.
- if they are not present in the `Context<S>` of the incoming request we would resort to the default
- DnsResolveIpMode has one more variant as there we also need to choose which one we are biased against, `Ipv4` or `Ipv6`, where the latter is the default to help progress the adoption of ipv6.

Sorry for not catching this earlier. Seens I missed it in my initial scans of your code.

Do let me know here or on Discord if something is not clear in that set of requirements.